uni/lib/view/common_widgets/toast_message.dart

```diff
@@ -1,17 +1,144 @@
-    ToastContext().init(context);
-    Toast.show(msg,
-        duration: Toast.lengthLong,
-        gravity: Toast.bottom,
-        backgroundColor: toastColor,
-        backgroundRadius: 16.0);
+class MToast extends StatelessWidget {
+  final String message;
+  final Color? color;
+  final IconData? icon;
+  final Color? iconColor;
+  final Color? textColor;
+  final AlignmentGeometry? alignment;
+  final dynamic elevation;
+
+  const MToast(
+      {Key? key,
+        required this.message,
+        this.color = Colors.white,
+        this.icon,
+        this.iconColor = Colors.black,
+        this.textColor = Colors.black,
+        this.alignment = Alignment.bottomCenter,
+        this.elevation = 0.0})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () async => false,
+      child: Dialog(
+        insetPadding: const EdgeInsets.all(50),
+        alignment: alignment,
+        backgroundColor: color,
+        shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(10.0))),
+        elevation: elevation,
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Container(
+                margin: const EdgeInsets.all(10.0),
+                child: icon != null ? Icon(
+                  icon,
+                  color: iconColor,
+                ) : const SizedBox.shrink()),
+            Expanded(
+              child: Text(
+                message,
+                style:
+                TextStyle(fontStyle: FontStyle.normal, color: textColor),
+                overflow: TextOverflow.ellipsis,
+                maxLines: 10,
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
+class ToastMessage{
+  static const Color toastErrorIconColor = Color.fromARGB(255, 241, 77 , 98),
+      toastErrorColor = Color.fromARGB(255, 252, 237 , 238),
+      toastSuccessIconColor = Color.fromARGB(255, 53, 210, 157),
+      toastSuccessColor = Color.fromARGB(255, 234, 250, 246),
+      toastWarningIconColor = Color.fromARGB(255, 244, 200, 98),
+      toastWarningColor = Color.fromARGB(255, 252, 244, 222),
+      toastInfoIconColor = Color.fromARGB(255, 40, 131, 229),
+      toastInfoColor = Color.fromARGB(255, 211, 229, 249);
+
+  static error(BuildContext context, String msg) {
+    return showDialog(
+        barrierDismissible: false,
+        barrierColor: Colors.white.withOpacity(0),
+        context: context,
+        builder: (_) {
+          Future.delayed(const Duration(milliseconds: 2000), () {
+            Navigator.of(context).pop();
+          });
+          return MToast(
+              message: msg,
+              color: toastErrorColor,
+              icon: CupertinoIcons.clear_circled_solid,
+              iconColor: toastErrorIconColor
+          );
+        });
+  }
```

Comment:
What about creating a new private method (ex: displayDialog) that receives a widget and returns the showDialog with that widget at Mtoast position and reuse code in the others methods